contrib/video_ffmpeg.lua

```diff
@@ -0,0 +1,448 @@
+  -> "It preserve {label} but substitute "
+```
+--]]
+local function format_string(label, symbols)
+  local es1, es2 = "\u{ffe0}", "\u{ffe1}" -- for simplicity, just some strange utf characters 
+  local result = label:gsub("\\{", es1):gsub("\\}", es2)
+  for s,v in pairs(symbols) do
+    result = result:gsub("{"..s.."}", v)
+  end
+  return result:gsub(es1, "{"):gsub(es2, "}")
+end
+
+-----  COMPONENTS
+
+local function combobox_pref_read(name, all_values)
+  local value = dt.preferences.read(MODULE_NAME, name, "string")
+  for i,v in pairs(all_values) do
+    if v == value then return i end
+  end
+  return 1
+end
+
+local function combobox_pref_write(name)
+  local writer = function(widget)
+    dt.preferences.write(MODULE_NAME, name, "string", widget.value)
+  end
+  return writer
+end
+
+local function string_pref_read(name, default)
+  local value = dt.preferences.read(MODULE_NAME, name, "string")
+  if value ~= nil and value ~= "" then return value end
+  return default
+end
+
+local function string_pref_write(name, widget_attribute)
+  widget_attribute = widget_attribute or "value"
+  local writer = function(widget)
+    dt.preferences.write(MODULE_NAME, name, "string", widget[widget_attribute])
+  end
+  return writer
+end
+
+local framerates_selector = dt.new_widget("combobox"){
+  label = _("framerate"),
+  tooltip = _("select framerate of output video"),
+  value = combobox_pref_read("framerate", framerates),
+  changed_callback = combobox_pref_write("framerate"), 
+  table.unpack(framerates)
+}
+
+local res_selector = dt.new_widget("combobox"){
+  label = _("resolution"),
+  tooltip = _("select resolution of output video"),
+  value = combobox_pref_read("resolution", res_list),
+  changed_callback = combobox_pref_write("resolution"),
+  table.unpack(res_list)
+}
+
+local codec_selector = dt.new_widget("combobox"){
+  label = _("codec"),
+  tooltip = _("select codec"),
+  value = combobox_pref_read("codec", codec_list),
+  changed_callback = combobox_pref_write("codec"),
+  table.unpack(codec_list)
+}
+
+local format_selector = dt.new_widget("combobox"){
+  label = _("format container"),
+  tooltip = _("select format of output video"),
+  value = combobox_pref_read("format", format_list),
+  changed_callback = function(widget)
+    combobox_pref_write("format")(widget)
+    codec_list = formats[widget.value]["codecs"]
+    table.sort(codec_list)
+    replace_combobox_elements(codec_selector, codec_list)
+  end,
+  table.unpack(format_list)
+}
+
+local destination_label = dt.new_widget("section_label"){
+  label = _("output file destination"),
+  tooltip = _("settings of output file destination and name")
+}
+
+local output_directory_chooser = dt.new_widget("file_chooser_button"){
+  title = _("Select export path"),
+  is_directory = true,
+  tooltip =_("select the target directory for the timelapse. \nthe filename is created automatically."),
+  value = string_pref_read("export_path", os.getenv("HOME")),
```

Comment:
One small improvement could be to set the default export directory to the Video folder and not home. Have a look into the "video_mencoder.lua" how we do it there:
https://github.com/darktable-org/lua-scripts/blob/master/contrib/video_mencoder.lua#L107

Reply:
good idea, I'll do that. I just need to figure out how to do it on Windows/Mac, I'm pretty sure `xdg-user-dir` wont work ;)

Reply:
On a mac it's $home/Movies

Reply:
hope it's more or less working now.

I've found that on Windows default video location can be changed to any directory you want. If I'm not mistaken, the only way to retrieve this information is to read registry. I don't know how frequent case it is. I'm open for contributions ;)